pkg/agent/dummy_data_gatherer.go

```diff
@@ -17,14 +18,25 @@ func (c *dummyConfig) NewDataGatherer(ctx context.Context) (datagatherer.DataGat
 		return nil, fmt.Errorf("an error")
 	}
 	return &dummyDataGatherer{
-		Param1: c.Param1,
+		Param1:         c.Param1,
+		FailedAttempts: c.FailedAttempts,
 	}, nil
 }
 
 type dummyDataGatherer struct {
-	Param1 string
+	Param1         string
+	attemptNumber  int
+	FailedAttempts int
 }
 
 func (c *dummyDataGatherer) Fetch() (interface{}, error) {
-	return nil, nil
+	var err error
+	if c.attemptNumber < c.FailedAttempts {
+		err = fmt.Errorf("First %d attempts will fail", c.FailedAttempts)
+	}
+	if c.Param1 == "foo" {
```

Comment:
do you really need `Param1` now? You can recreate the failure just with `FailedAttemps`, don't you?

Reply:
Yes, you could set failedAttempts to a very large number and Param1 would become redundant. I didn't add it in however, so I'm not sure if its there for some other reason

Reply:
if we want a "fail always" option, I would rename `Param1` to `FailAlways`